Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1

```diff
@@ -9,16 +9,12 @@ Param(
 )
 
 function DownloadDependenciesFromProbingPaths($baseFolder, $project, $destinationPath) {
-    $projectSettings = ReadSettings -baseFolder $baseFolder -project $project
-    
-    $probingPaths = $projectSettings.appDependencyProbingPaths | ConvertFrom-Json
-
-    $downloadedDependencies = @()
-    if ($probingPaths) {
-        $downloadedDependencies += Get-Dependencies -probingPathsJson $repo.appDependencyProbingPaths -saveToPath $destinationPath  | Where-Object { $_ }
+    $settings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable -recurse
+    $settings = AnalyzeRepo -settings $settings -token $token -baseFolder $baseFolder -project $project -doNotCheckArtifactSetting
+    if ($settings.ContainsKey('appDependencyProbingPaths') -and $settings.appDependencyProbingPaths) {
+        return Get-Dependencies -probingPathsJson $settings.appDependencyProbingPaths -saveToPath $destinationPath | Where-Object { $_ }
     }
-
-    return $downloadedDependencies
+    return $settings
```

Comment:
So, the function returns either the settings or the result from `Get-Dependencies`? I don't get why...

Reply:
Bug - I was fixing that while you were reviewing - sorry